expr.go

```diff
@@ -293,23 +319,40 @@ func getStringArg(e *expr, n int) (string, error) {
 		return "", ErrMissingArgument
 	}
 
-	if e.args[n].etype != etString {
-		return "", ErrBadType
+	return doGetStringArgDefault(e.args[n], "")
+}
+
+func getStringArgDefault(e *expr, n int, s string) (string, error) {
+	if len(e.args) <= n {
+		return s, nil
 	}
 
-	return e.args[n].valStr, nil
+	return doGetStringArgDefault(e.args[n], s)
 }
 
-func getStringArgDefault(e *expr, n int, s string) (string, error) {
+func getStringNamedOrPosArgDefault(e *expr, k string, n int, s string) (string, error) {
 	if len(e.args) <= n {
```

Comment:
Indeed. The length check is only needed when we know the argument is not a named one. I'll move it from `doGetTypeNamedOrPosArgument` to `getTypeArgDefault`.